### ISSUE TITLE
Type error message

### ISSUE THREAD

**Issue by ulugbekna** [ocaml/ocaml-lsp#109]
Current type mismatch message is hard to parse, e.g., on VS Code 

<img width="737" alt="Screen Shot 2020-02-29 at 3 04 24 PM" src="https://user-images.githubusercontent.com/16353531/75609621-79d4a100-5b0a-11ea-9db2-43fdef62789f.png">

and, especially, with larger types.

What do people think about a shorter (and formatted with markdown) message? For example, Intellij IDEA for Scala:

<img width="290" alt="Screen Shot 2020-02-29 at 3 02 22 PM" src="https://user-images.githubusercontent.com/16353531/75609705-3169b300-5b0b-11ea-83e7-5288cf5324ea.png">

and

<img width="515" alt="Screen Shot 2020-02-29 at 3 03 01 PM" src="https://user-images.githubusercontent.com/16353531/75609735-5b22da00-5b0b-11ea-84ee-76fea7658c00.png">

(might not be the best place to fix an error message, but this project seems more flexible to changes)

**Comment by rgrinberg** [comment]
It would be obviously nice to have this, but isn't there work in the compiler to provide structured type errors already? Ideally, we would just piggy back on that and format those structured errors as markdown.

Let's see what @trefis or @let-def recommend.

**Comment by let-def** [comment]
When merlin catches an error from the compiler it is already reasonably structured.
In this case we could pattern match on `Expr_type_clash (...)` and provide an alternative formatting. But adding more special cases is not really satisfying either.

**Comment by trefis** [comment]
I believe what @rgrinberg was referring to are some discussions from last summer (or the previous one?) about the compiler exposing errors as json objects, or something of the sort.

AFAICT no work has been done on that front, but I don't think there would be a strong opposition if you submitted a PR upstream (i.e. in the compiler) that implements that.
Then merlin would get it for free (and we could backport it to older versions).
